core/vm/contracts_test.go

```diff
@@ -134,17 +102,7 @@ var vmctx = Context{
 		if header.Number.Cmp(testHeader.Number) > 0 {
 			return false
 		}
-
-		// FIXME: Implementation currently relies on the Istanbul engine's internal view of the
-		// chain, so return false if this is not an Istanbul chain. As a consequence of this the
-		// seal is always verified against the canonical chain, which makes behavior undefined if
-		// this function is evaluated on a chain which does not have the highest total difficulty.
-		if chainCtx.Config().Istanbul == nil {
-			return false
-		}
-
-		// Submit the header to the engine's seal verification function.
-		return chainCtx.Engine().VerifySeal(nil, header) == nil
+		return true
```

Comment:
I kind of want to ask what the deal was with the `return chainCtx.Engine().VerifySeal(nil, header) == nil`, but then again I probably don't want to know.

Reply:
lol... it's not that bad. This test used a mock engine where VerifySeal is implemented as `return nil`, so doing `nil == nil` is alwas true